measurecrack.py

```diff
@@ -0,0 +1,175 @@
+    
+    # Find max area contour
+    maxArea = 0
+    
+    for c in contours:
+        if cv2.contourArea(c) > maxArea:
+            maxArea = cv2.contourArea(c)
+            maxc = c
+    
+    # Find minimum bounding rectangle
+    rect = cv2.minAreaRect(maxc)
+    width = rect[1][0]
+    height = rect[1][1]
+    return width, height
+
+# Measures crack using perspective transformation method
+def measure_crack2(image):
+    # define range of color in HSV
+    lower = np.array([0,0,0])
+    upper = np.array([180,50,100])
+
+    #Gaussian Blur
+    blurredcolor = cv2.GaussianBlur(image, (5, 5), 0)
+    
+    # Convert BGR to HSV
+    hsv = cv2.cvtColor(blurredcolor, cv2.COLOR_BGR2HSV)
+    
+    # Threshold the HSV image to get only black colors
+    mask = cv2.inRange(hsv, lower, upper)
+    
+    # resized = imutils.resize(mask, width=600)
+    # cv2.imshow("mask", resized)
+    # cv2.waitKey(0)
+    
+    # Detect lines
+    lines = cv2.HoughLines(mask, 1, np.pi/180, 700)
+    # If no lines are detected, revert to ratio method
+    if (lines is None):
+        return measure_crack(image)
+    a,b,c = lines.shape
+    
+    top = []
+    bottom = []
+    left = []
+    right = []
+    
+    height, width, channels = image.shape
+    
+    for i in range(a):
+        rho = lines[i][0][0]
+        theta = lines[i][0][1]
+        
+        # Sort lines into left, right, top, and bottom
+        if (theta < 0.2):
+            if (rho < 0.25 * width):
+                left.append([rho, theta])
+            elif (rho > 0.75 * width):
+                right.append([rho, theta])
+        elif (theta > 2.94):
+            if (rho > -0.25 * width):
+                left.append([rho, theta])
+            elif (rho < -0.75 * width):
+                right.append([rho, theta])
+        elif (theta > 1.37 and theta < 1.77):
+            if (rho < 0.25 * height):
+                top.append([rho, theta])
+            elif (rho > 0.75 * height):
+                bottom.append([rho, theta])
+    
+    # If grid line detection fails, revert to ratio method
+    if (len(top) == 0 or len(bottom) == 0 or len(left) == 0 or len(right) == 0):
+        return measure_crack(image)
+
+    x1, y1 = intersect(top, left)
+    x2, y2 = intersect(top, right)
+    x3, y3 = intersect(bottom, left)
+    x4, y4 = intersect(bottom, right)
+    
+    # Draw lines on image
+    # cv2.line(image, (x1, y1), (x2, y2), (0, 255, 0), 5)
+    # cv2.line(image, (x1, y1), (x3, y3), (0, 255, 0), 5)
+    # cv2.line(image, (x2, y2), (x4, y4), (0, 255, 0), 5)
+    # cv2.line(image, (x3, y3), (x4, y4), (0, 255, 0), 5)
+    
+    # Display image with lines drawn
+    # resized = imutils.resize(image, width=600)
+    # cv2.imshow("lines", resized)
+    # cv2.waitKey(0)
+    
+    # Find perspective transform matrix
+    src = np.float32([[x1, y1], [x2, y2], [x3, y3], [x4, y4]])
+    dst = np.float32([[0, 0], [3000, 0], [0, 3000], [3000, 3000]])
```

Comment:
no magic numbers

Reply:
here or in the rest of the file, idk why i decided to leave all these comments instead of just making a note at the top, but i dont feel like going back and removing them at this point